### PR TITLE
Pandas assignment

### PR DESCRIPTION
**Pull request by DaliJebali**

### CODE REVIEW

**Comment by tomMoral** [comment]
If you are finished, you can sign up for this RAMP challenge: https://ramp.studio/events/brevet_datacamp2020
This is what we are going to do this afternoon. The goal is to try and solve the challenge.
You need to click on the blue button: `Join the event` and I need to approve your participation so send me a message once you have done this.
You can then read the starting kit and go to the left menu to submit your solutions in the sandbox.

File: .idea/datacamp-assignment-pandas.iml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="PYTHON_MODULE" version="4">
+  <component name="NewModuleRootManager">
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="jdk" jdkName="Python 3.7" jdkType="Python SDK" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (tomMoral) [inline .idea/datacamp-assignment-pandas.iml:8]: Don't add unrelated files to the PR.

File: .idea/deployment.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      <paths name="ajebali@ssh.enst.fr:22">
+        <serverdata>
+          <mappings>
+            <mapping local="$PROJECT_DIR$" web="/" />
+          </mappings>
+        </serverdata>
+      </paths>
+    </serverData>
+  </component>
+</project>
```
Comment (tomMoral) [inline .idea/deployment.xml:14]: ditto

File: pandas_questions.py [hunk 3]
```diff
@@ -28,8 +28,20 @@ def merge_regions_and_departments(regions, departments):
...
+    )
+    # merge on region code
 
-    return pd.DataFrame({})
+    return pd.merge(
+        reg,
+        dep,
+        left_on="code_reg",
+        right_on="region_code"
+    ).drop(['region_code'], axis=1)
```
Comment (tomMoral) [inline pandas_questions.py:44]: RMK: you can use `suffixes` in merge to rename the questions.

File: pandas_questions.py [hunk 4]
```diff
@@ -38,8 +50,22 @@ def merge_referendum_and_areas(referendum, regions_and_departments):
     You can drop the lines relative to DOM-TOM-COM departments, and the
     french living abroad.
     """
+    # Pad Department code from 1 to 01
+    referendum['Department code'] = referendum['Department code'].apply(
+        lambda x: x.zfill(2)
+    )
```
Comment (tomMoral) [inline pandas_questions.py:56]: You can directly use `df.str.zfill` instead of doing apply.

File: pandas_questions.py [hunk 5]
```diff
@@ -38,8 +50,22 @@ def merge_referendum_and_areas(referendum, regions_and_departments):
...
 
-    return pd.DataFrame({})
+    # lst = ['01', '02', '03', '04', '06', 'COM']
+    # ref2=referendum.copy()
+    # indexes = ref2[ref2['Department code'].isin(lst)].index
+    # ref2.drop(indexes, inplace=True)
```
Comment (tomMoral) [inline pandas_questions.py:61]: Remove old code comments in the PR.

**Review by tomMoral — CHANGES_REQUESTED** [review]
Mostly good but please remove the extra file from the pr.

File: .idea/vcs.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (tomMoral) [inline .idea/vcs.xml:6]: Il faudrait que tu enleves tous les fichiers dans `.idea`.
Je pense que c'est `git rm --cached -r .idea`

**Review by tomMoral — CHANGES_REQUESTED** [review]
still extra files.
You can look at the file changes tab in the PR.

File: .idea/vcs.xml (hunk 6 above)
Comment (DaliJebali) [reply .idea/vcs.xml:6]: So I run git rm --cached -r .idea and then commit again?

**Review by tomMoral — APPROVED** [review]
LGTM!

File: .idea/vcs.xml (hunk 6 above)
Comment (tomMoral) [reply .idea/vcs.xml]: yep! thx!
